### PR TITLE
Add support for generating cloud-init network-config

### PR DESCRIPTION
**Pull request by greg-bock**
**What this PR does / why we need it**:
Add support for generating cloud-init network-config

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Add support for generating cloud-init network-config
```

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by greg-bock** [comment]
This PR is the network-config pieces minus SR-IOV from #1806 
@davidvossel @karmab

**Comment by greg-bock** [comment]
Documentation in https://github.com/kubevirt/user-guide/pull/189

File: pkg/virt-api/webhooks/validating-webhook/validating-webhook_test.go [hunk 1]
```diff
@@ -991,6 +993,44 @@ var _ = Describe("Validating Webhook", func() {
 	})
 
 	Context("with VirtualMachineInstance spec", func() {
+		It("should accept valid machine type", func() {
```
Comment (davidvossel) [inline pkg/virt-api/webhooks/validating-webhook/validating-webhook_test.go]: There's a lot more test cases than just the network-config tests being added here. Is this intentional? I reviewed them and they look good, it was just unexpected to see these additional validation tests in this patch.

Comment (greg-bock) [reply pkg/virt-api/webhooks/validating-webhook/validating-webhook_test.go]: Once i saw coveralls I felt the need to go green... I was already adding some missing tests so I figured why not a few more?

**Comment by greg-bock** [comment]
> The code looks great to me.
> 
> The only think missing is a functional test that verifies the network-config works.
> 
> Take a look at `tests/vmi_userdata_test.go`. We have functional tests there that verify user data can both be added to a VMI and that the user data is picked up by the VMI when it is launched. 

I had tests when SRIOV was part of the PR in the vmi_multus_test. I missed porting any of that and will try to get it done this weekend.

**Comment by greg-bock** [comment]
Tests added, can I get a full CI test?

**Comment by SchSeba** [comment]
Hi @greg-bock can you please take a look on the travis failed?
```
------------------------------
• Failure [0.003 seconds]
Virt-api
/root/go/src/kubevirt.io/kubevirt/pkg/virt-api/api_test.go:51
  Virt api server
  /root/go/src/kubevirt.io/kubevirt/pkg/virt-api/api_test.go:206
    should not found if endpoint does not exists [It]
    /root/go/src/kubevirt.io/kubevirt/pkg/virt-api/api_test.go:396
    Expected
        <int>: 200
    to equal
        <int>: 404
    /root/go/src/kubevirt.io/kubevirt/pkg/virt-api/api_test.go:405
------------------------------
•••••the server could not find the requested resource (get validatingwebhookconfigurations.admissionregistration.k8s.io virt-api-validator)
•••••••••••••••••••
Summarizing 1 Failure:
[Fail] Virt-api Virt api server [It] should not found if endpoint does not exists 
/root/go/src/kubevirt.io/kubevirt/pkg/virt-api/api_test.go:405
```

**Comment by greg-bock** [comment]
This is an intermittent error in testing as far as I know, rerun the test.

**Comment by greg-bock** [comment]
It is failing on other PR builds too.

**Comment by greg-bock** [comment]
I can't dupe this in my environment and nothing in my last two commits (mostly functest additions and a log output change) should have made it start failing from the previous commit(https://github.com/kubevirt/kubevirt/pull/1923/commits/4a277f0f2ff054d65d9d070c101ee519b3870292).

**Comment by slintes** [comment]
the travis failure will be fixed by https://github.com/kubevirt/kubevirt/pull/1935
will retrigger your build after merge

**Comment by slintes** [comment]
ci test please

**Comment by greg-bock** [comment]
Thanks @slintes

**Comment by davidvossel** [comment]
ci test please

**Comment by davidvossel** [comment]
ci test please

**Comment by davidvossel** [comment]
@greg-bock thanks for the ping, sorry this pr didn't get merged. I've rebased the code and kicked of ci once again.

**Comment by greg-bock** [comment]
Thanks!

**Pull request merged**